crypto/store/store_lib.c

```diff
@@ -932,9 +932,17 @@ OSSL_STORE_CTX *OSSL_STORE_attach(BIO *bp, const char *scheme,
 
     OSSL_TRACE1(STORE, "Looking up scheme %s\n", scheme);
 #ifndef OPENSSL_NO_DEPRECATED_3_0
+    ERR_set_mark();
     if ((loader = ossl_store_get0_loader_int(scheme)) != NULL)
         loader_ctx = loader->attach(loader, bp, libctx, propq,
                                     ui_method, ui_data);
+    /*
+     * ossl_store_get0_loader_int will will raise an error if the loader for
+     * the scheme cannot be retrieved. This error is cleared as the code below
+     * will try to fetch the loader from the provider.
+     */
+    if (loader == NULL)
+      ERR_pop_to_mark();
```

Comment:
You have to also call ERR_clear_last_mark() in the else branch.

Reply:
Or, alternatively, you can call the ERR_pop_to_mark() unconditionally as there should be no error on the stack anyway in the else case.

Reply:
Thanks, I've updated with your suggestion now.